lib/tests/streamlit/runtime/caching/common_cache_test.py

```diff
@@ -463,6 +464,22 @@ def foo(i):
             st.text("---")
             foo(1)
 
+    @parameterized.expand([("memo", memo), ("singleton", singleton)])
+    def test_cached_st_image_replay(self, _, cache_decorator):
```

Comment:
I get that we are calling the img twice for image replay but is there any assertions that should be called? It feels like this test is a little odd. 

I don't quite understand it so maybe there should be some comments explaining how to the test works.

Reply:
Should we maybe be asserting that some call to caching was called (sorry I'm not a caching expert even though I just attended the SLOOSH session that talked about caching)

Reply:
I'll add some comments. The test is to catch any trivial immediate errors thrown by calling `image` in a cached function, or playing it from cache. Actual caching behavior/functionality is much harder to verify usefully in these tests, so I left that to the e2e tests.

Reply:
SGTM